<launch>
        <!-- Desired frequency to play the dataset, POSSIBLE VALUES ARE (30, 15, 10) Hz -->
        <arg name="label_folder_arg" default="label_nyu40" />
        <arg name="label_colored_folder_arg" default="label_nyu40_colored" />
        <arg name="frequency_arg" default="30" />
        <arg name="scene_id_arg" default="scene0000_00" />
        <arg name="first_message_number_arg" default="0" />
        <arg name="scannet_folder_arg" default="/root/scannet" />

        <!-- Desired dimensions of depth images, color images are scaled accordingly --> 

        <arg name="W_depth_arg" default="640" />
        <arg name="H_depth_arg" default="480" />

        <node pkg="ros_visual_datasets" type="scannet_to_ros.py" name="scannet_to_ros" output="screen">
            <param name="label_folder" value="$(arg label_folder_arg)" />
            <param name="label_colored_folder" value="$(arg label_colored_folder_arg)" />
            <param name="frequency" value="$(arg frequency_arg)" />
            <param name="first_message_number" value="$(arg first_message_number_arg)" />
            <param name="scannet_folder" value="$(arg scannet_folder_arg)" />
            <param name="scene_id" value="$(arg scene_id_arg)" />
            
            <param name="W_depth" value="$(arg W_depth_arg)" />
            <param name="H_depth" value="$(arg H_depth_arg)" />

        </node>
        <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ros_visual_datasets)/rviz/scannet.rviz" />

  <node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager">
    <remap from="camera_info" to="/camera/depth/camera_info"/>
    <remap from="image_rect" to="/camera/depth/image_raw"/>
    <remap from="points" to="/camera/depth/points"/>
  </node>
</launch>